buildsystem/modules/FindPython.cmake

```diff
@@ -112,13 +126,9 @@ endforeach()
 foreach(PYTHON ${PYTHON_INTERPRETERS})
 	# ask the interpreter for the essential extension-building flags
 	py_get_config_var(INCLUDEPY PYTHON_INCLUDE_DIR)
-	if(APPLE)
-		py_get_config_var(DESTLIB PYTHON_LIBRARY_PATH)
-		get_filename_component(PYTHON_LIBRARY ${PYTHON_LIBRARY_PATH} NAME)
-		get_filename_component(PYTHON_LINK_DIRECTORIES ${PYTHON_LIBRARY_PATH} DIRECTORY)
-	else()
-		py_get_config_var(BLDLIBRARY PYTHON_LIBRARY)
-	endif()
+	py_get_config_var(LIBPL PYTHON_LIBRARY_DIR)
+	py_get_lib_name(PYTHON_LIBRARY_NAME)
+    set(PYTHON_LIBRARY "-L${PYTHON_LIBRARY_DIR} -l${PYTHON_LIBRARY_NAME}")
```

Comment:
D'oh! Good catch. Old habits die hard... I'll check in the change with the squash.